src/uct/ib/base/ib_iface.c

```diff
@@ -888,7 +896,10 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
     if (uct_ib_iface_is_roce(self) || config->is_global ||
         /* check ADDR_TYPE for backward compatibility */
         (config->addr_type == UCT_IB_ADDRESS_TYPE_SITE_LOCAL) ||
-        (config->addr_type == UCT_IB_ADDRESS_TYPE_GLOBAL)) {
+        (config->addr_type == UCT_IB_ADDRESS_TYPE_GLOBAL) ||
+        /* check subnet prefix for IB only */
+        (uct_ib_iface_is_ib(self) &&
```

Comment:
due to `uct_ib_iface_is_roce` check before? I would not assume that `uct_ib_iface_is_ib == !uct_ib_iface_is_roce` because:
1. 3d party lib defines this enum and we can have another link layer in the next versions of it
2. we define separated constants for that in UCX config.h by `configure`

Reply:
ok